client/redux/cart/actions.js

```diff
@@ -0,0 +1,137 @@
+import {
+  TOGGLE_CART,
+  UPDATE_CART,
+  FETCH_CART_PRODUCTS_FAILURE,
+  FETCH_CART_PRODUCTS_SUCCESS,
+  FETCH_CART_PRODUCTS_REQUEST,
+  UPDATE_CART_ID,
+} from './constants'
+
+import * as api from '../../api/cart'
+
+export const toggleCart = () => ({
+  type: TOGGLE_CART,
+})
+
+export const updateCart = (cart) => ({
+  type: UPDATE_CART,
+  payload: cart,
+})
+
+export const fetchCartProductsRequest = () => ({
+  type: FETCH_CART_PRODUCTS_REQUEST,
+})
+
+export const fetchCartProductsFailure = (error) => ({
+  type: FETCH_CART_PRODUCTS_FAILURE,
+  payload: error,
+})
+
+export const fetchCartProductsSuccess = (products) => ({
+  type: FETCH_CART_PRODUCTS_SUCCESS,
+  payload: products,
+})
+
+export const getCartProducts = () => async (dispatch, getState) => {
+  const { id: cartId } = getState().cart
+  dispatch(fetchCartProductsRequest())
+
+  try {
+    const data = await api.getCart(cartId)
+
+    const { products, total } = await data.data
+
+    const quantity = products.reduce((accum, product) => (accum += product.quantity), 0)
```

Comment:
@AStashevskaya Good practice in using `reduce` - not modify `accumulator` variable, and use it in calculation instead, try 

```js
const quantity = products.reduce((accum, product) => accum + product.quantity, 0)
```